website/integrations: Added Tandoor Recipe documentation (#6609)
goauthentik/authentik: website/integrations/services/tandoor-recipes/index.md

---
title: Tandoor Recipes
---

<span class="badge badge--secondary">Support level: Community</span>

## What is Tandoor Recipes

> Tandoor Recipes is an easy-to-use, powerful self-hosted food recipe manager.
>
> -- https://docs.tandoor.dev/

Tandoor Recipy uses Django Allauth as the authentication engine. It supports OpenID Connect.

## Preparation

The following placeholders will be used:

-   `tandoor.company` is the FQDN of the Tandoor Recipes install.
-   `authentik.company` is the FQDN of the authentik install.

### Provider

> Create an Provider in authentik with the following parameters:

Reviewer: ```suggestion
Create a Provider in authentik with the following parameters:
```

-   Client type

    Confidential

-   Client ID

>     Take note off Client ID.

Reviewer: ```suggestion
    Take note of the Client ID.
```

-   Client Secret

>     Take note off Client Secret.

Reviewer: ```suggestion
    Take note of the Client Secret.
```

-   Redirect URIs/Origins (RegEx)

>     https://tandoor.company/accounts/authentik/login/callback/

Reviewer 2: This should be `https://tandoor.company/accounts/oidc/authentik/login/callback/` (note the added 'oidc').

Reviewer 3: Were you able to fix the issue with linking existing local accounts to OIDC accounts?

Reviewer 2: Yes. I configured it with Authentik as per the instructions in this PR (with the change as per my previous comment in the callback URL). I logged in as my local Tandoor user, went into user settings and there was a button to connect a social account. I did so, it went through Authentik's authorisation flow, and redirected me back to Tandoor, at which time it told me it was successfully linked. I've since logged out and back in without issue.

Reviewer 3: figured out that I already got the 'oidc' part in my CallbackURL but it is still erroring out with Server Error 500 (ofc most recent version, both)

Reviewer 3: I found the issue :/

I had to convert the PSQL encodging from SQL_ASCII to UTF8 on tandoor and after this my account was able to be connected with authentik

Reviewer 4: Ok I think i figured it out. I had both configured - OIDC within the Django admin panel & as environment variables. I guess it was related to previous experiments. However it worked fine for the last 10 months. 
Now I removed the config from Django admin and exclusively configured docker env variables.
I detached the identities from the old OIDC provider and linked it again - now it is working again.

-   Signing Key

    authentik Self-signed Certificate

> -   Leave the rest on default

Reviewer: ```suggestion
-   Leave the rest as the default options.
```

### Application

> Create a Application with the following parameters:

Reviewer: ```suggestion
Create an Application with the following parameters:
```

-   Name: tandoor

-   Provider: tandoor

> -   Leave rest on default

Reviewer: ```suggestion
-   Leave the rest as the default options.
```

> ## Tandoor Recipy

Reviewer: ```suggestion
## Tandoor Recipe
```

> -   Set the following environment variables on your Tandoor Recipes installation (please make sure to use the correct values for `server_url`, `client_id` and `secret`):

Reviewer: @fyksen heads up about above note...

-   Restart Tandoor, you should now see `Sign In Via Authentik` button.

> -   If you want to set your authentik user as administrator, log in as an administrator, then you can do so in the https://tandoor.company/admin/auth/user/ click the user you want to make admin, and mark the `Superuser status` and `Staff status`.

Reviewer: ```suggestion
-   If you want to set your authentik user as the administrator, log in as an administrator. Then navigate to  `https://tandoor.company/admin/auth/user/`, click the user you want to make an administrator, and select the `Superuser status` and `Staff status`.
```

---
title: Tandoor Recipes
---

<span class="badge badge--secondary">Support level: Community</span>

## What is Tandoor Recipes

> Tandoor Recipes is an easy-to-use, powerful self-hosted food recipe manager.
>
> -- https://docs.tandoor.dev/

> Tandoor Recipy uses Django Allauth as the authentication engine. It supports Generic OATH 2, openid_connect.

Reviewer 5: ```suggestion
Tandoor Recipy uses Django Allauth as the authentication engine. It supports OpenID Connect.
```

## Preparation

The following placeholders will be used:

> -   `tandoor.company` is the FQDN of the Uptime Kuma install.

Reviewer 5: ```suggestion
-   `tandoor.company` is the FQDN of the Tandoor Recipes install.
```

-   `authentik.company` is the FQDN of the authentik install.

### Provider

Create an Provider in authentik with the following parameters:

-   Client type

    Confidential

-   Client ID

    Take note off Client ID.

-   Client Secret

    Take note off Client Secret.

-   Redirect URIs/Origins (RegEx)

    https://tandoor.company/accounts/authentik/login/callback/

-   Signing Key

    authentik Self-signed Certificate

-   Leave the rest on default

### Application

Create a Application with the following parameters:

-   Name: tandoor

-   Provider: tandoor

-   Leave rest on default

## Tandoor Recipy

> -   Add this to your `.env` file (please make sure to insert your `server_url`, `client_id` and `secret`):
> 
> ```
> SOCIALACCOUNT_PROVIDERS={ "openid_connect": { "SERVERS": [{ "id": "authentik", "name": "Authentik", "server_url": "https://authentik.company/application/o/tandoor/.well-known/openid-configuration", "token_auth_method": "client_secret_basic", "APP": { "client_id": "ClientIdFromProvider", "secret": "SecretFromProvider" } } ] } }
> ```

Reviewer 5: ```suggestion
-   Set the following environment variables on your Tandoor Recipes installation (please make sure to use the correct values for `server_url`, `client_id` and `secret`):
```

(GitHub is dying with this suggestion because of the backticks, you probably know what I mean)
